### PR TITLE
fix: version is a required argument, fix CRDv1

### PR DESCRIPTION
**Pull request by evadnoob**
updates `ok.CRDv1()`, version is pretty essential, and required argument. 

Also updates the PodDisruptionBudget version

### CODE REVIEW

File: kubernetes/kube.libsonnet [hunk 1]
```diff
@@ -144,7 +144,7 @@
       spec: {
         group: group,
         names: names,
-        versions: error 'versions required',
+        version: version,
```
Comment (jkinkead) [inline kubernetes/kube.libsonnet:147]: Shouldn't this be `versions`, per [the API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#customresourcedefinitionspec-v1-apiextensions-k8s-io)?

Comment (evadnoob) [reply kubernetes/kube.libsonnet]: Yes, I had local changes, pushed now.

File: kubernetes/kube.libsonnet [hunk 2]
```diff
@@ -703,7 +703,7 @@
 
   LimitRange(name, namespace): $._Object('v1', 'LimitRange', name, namespace=namespace),
 
-  PodDisruptionBudget(name, namespace, app=name): $._Object('policy/v1beta1', 'PodDisruptionBudget', name, namespace=namespace) {
+  PodDisruptionBudget(name, namespace, app=name): $._Object('policy/v1', 'PodDisruptionBudget', name, namespace=namespace) {
```
Comment (jkinkead) [inline kubernetes/kube.libsonnet:706]: Add to PR description?

This is needed for the 1.25 upgrade, if I'm reading things correctly, and should be available starting at 1.21.

Is this function used in any 1.14 clusters, like the old ops cluster?

Comment (evadnoob) [reply kubernetes/kube.libsonnet:706]: The conditionality of this argument is actually more difficult to fix than it looks initially. 

I thought about making a conditional or variable argument to the function and default to `v1beta1`, with an override for modern deployments. But I am thinking to do the opposite now. Default to `policy/v1` and for deployments that go to ops clusters they will need to be updated to override to `policy/v1beta1`

Comment (evadnoob) [reply kubernetes/kube.libsonnet]: I pushed an update for jsonnet-libs which I think works for us. But it makes the tradeoff to move forward and requires us to modify backward for ops cluster(ops.us-west-2). I’ll identify which apps will need updates and follow up on those.

**Review by AndrewWinterman — APPROVED** [review]

File: kubernetes/kube.libsonnet [hunk 3]
```diff
@@ -703,7 +702,8 @@
 
   LimitRange(name, namespace): $._Object('v1', 'LimitRange', name, namespace=namespace),
 
-  PodDisruptionBudget(name, namespace, app=name): $._Object('policy/v1beta1', 'PodDisruptionBudget', name, namespace=namespace) {
+  PodDisruptionBudget(name, namespace, app=name, version='policy/v1'): $._Object( version,
```
Comment (masterkidan) [inline kubernetes/kube.libsonnet:705]: Looks like this is the only change
```
an empty spec.selector ({}) written to a policy/v1 PodDisruptionBudget selects all pods in the namespace (in policy/v1beta1 an empty spec.selector selected no pods). An unset spec.selector selects no pods in either API version.
```
Probably just safe to update in place as we always specify a selector

**Review by masterkidan — APPROVED** [review]
LGTM

File: kubernetes/kube.libsonnet (hunk 3 above)
Comment (evadnoob) [reply kubernetes/kube.libsonnet:705]: Agree, probably ok everywhere but possibly/likely breaks ops clusters

File: kubernetes/kube.libsonnet (hunk 2 above)
Comment (jkinkead) [reply kubernetes/kube.libsonnet]: This seems overly-aggressive to me, given that this upgrade isn't happening for at least a quarter. Why not just wait until the ops cluster is retired, and save yourself some work?

Comment (evadnoob) [reply kubernetes/kube.libsonnet]: Ok, I was not thinking about the timeline, I can take out the PodDisruptionBudget change

**Review by masterkidan — APPROVED** [review]
LGTM

**Pull request merged**